File: app/src/main/java/me/imzack/app/ender/view/contract/HomeViewContract.java
package me.imzack.app.ender.view.contract;

public interface HomeViewContract extends BaseViewContract {

    void showInitialView(String planCount, int textSize, String planCountDscpt);

    void changePlanCount(String planCount, int textSize);

    void changeDrawerHeaderDisplay(String planCount, int textSize, String planCountDscpt);

    void closeDrawer();

    void showFragment(String tag);

    void onPressBackKey();

    void enterActivity(String tag);

    void showToast(String msg);
}

File: app/src/main/kotlin/me/imzack/app/end/view/contract/HomeViewContract.kt
package me.imzack.app.end.view.contract

interface HomeViewContract : BaseViewContract {

    fun showInitialView(planCount: String, textSize: Int, planCountDscpt: String)

    fun showInitialFragment(restored: Boolean, shownTag: String)

    fun switchFragment(fromTag: String, toTag: String)

    fun changePlanCount(planCount: String, textSize: Int)

    fun changeDrawerHeaderDisplay(planCount: String, textSize: Int, planCountDscpt: String)

    fun closeDrawer()

    fun onPressBackKey()

    fun startActivity(tag: String)

    fun showToast(msg: String)
}
